src/components/Form.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  InputAdornment,
+  FormControl,
+  InputLabel,
+  OutlinedInput,
+  useMediaQuery,
+  Typography,
+} from "@mui/material";
+import SearchIcon from "@mui/icons-material/Search";
+import { useTheme } from "@mui/material/styles";
+import { useForm, SubmitHandler } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+
+export interface IAppProps {}
+
+const schema = yup
+  .object({
+    city: yup
+      .string()
+      .required("Looks like you forgot to type a city you are interested in")
+      .test("regex test", `Numbers can't be in a city name`, (val) => {
+        const regExp = /^[a-zA-Z\s]+$/;
+        return regExp.test(val as string);
+      })
+      .matches(/^[a-zA-Z\s]+$/, `Numbers can't be in a city name`),
+  })
+  .required();
+
+type FormData = yup.InferType<typeof schema>;
+
+type Mode = {
+  onTouched: "onTouched";
+};
+
+export default function Form(props: IAppProps) {
+  const theme = useTheme();
+  const matches: boolean = useMediaQuery<string | undefined>(
+    theme.breakpoints.up("sm")
+  );
+  const {
+    register,
+    reset,
+    handleSubmit,
+    clearErrors,
+    formState: { errors },
+  } = useForm<FormData>({
+    mode: "onTouched",
+    resolver: yupResolver(schema),
+  });
+  const handleFormSubmit: SubmitHandler<FormData | Mode> = (values) => {
+    console.log(values);
```

Comment:
It's considered a good practice to remove console logs before committing to master